src/entity/entity_types.cpp

```diff
@@ -94,3 +98,104 @@ EntityType type_of(SoundEntity *e) {
 /*
  * End of `tools/entity_types_type_of.cpp`
  */
+
+/*
+ * Included from `tools/entity_types_event_callback.cpp`
+ */
+
+void EventCreateEntity::callback() {
+    switch (type) {
+        case EntityType::BASEENTITY: {
+            BaseEntity entity;
+            std::memcpy(&entity, BASEENTITY, sizeof(BaseEntity));
+            GAMESTATE()->entity_system.add(entity);
+            break;
+        }
+        case EntityType::ENTITY: {
+            Entity entity;
+            std::memcpy(&entity, ENTITY, sizeof(Entity));
+            GAMESTATE()->entity_system.add(entity);
+            break;
+        }
+        case EntityType::LIGHT: {
+            Light entity;
+            std::memcpy(&entity, LIGHT, sizeof(Light));
+            GAMESTATE()->entity_system.add(entity);
+            break;
+        }
+        case EntityType::PLAYER: {
+            Player entity;
+            std::memcpy(&entity, PLAYER, sizeof(Player));
+            GAMESTATE()->entity_system.add(entity);
+            break;
+        }
+        case EntityType::SOUNDENTITY: {
+            SoundEntity entity;
+            std::memcpy(&entity, SOUNDENTITY, sizeof(SoundEntity));
+            GAMESTATE()->entity_system.add(entity);
+            break;
+        }
+
+        default:
+            UNREACHABLE("Unknown entity type");
+            break;
+    }
+}
```

Comment:
This won't work when we introduce networking. You will need to explicitly overwrite the V-Table pointer for all entities.

You would implement this by (I would recomend) not copying the first sizeof(void *) bytes over the entity. Or you could explicitly set those bytes to the correct addresses. Due to user input not being determenistic, we can't assume the V-Table pointers are in the right place.